src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -68,7 +68,7 @@ public function createQueryBuilderByChannelAndTaxonSlug(ChannelInterface $channe
             ->andWhere('channel = :channel')
             ->andWhere('o.enabled = true')
             ->andWhere('channelPricing.channel = :channel')
-            ->groupBy('o.id')
+            ->groupBy('o.id, translation.id, productTaxon.position')
```

Comment:
I've just checked available options on current master... We also have a sorting by price.